### ISSUE TITLE
HTML navigation: Is there a way to determine a specific text to be read for a certain <div>?

### ISSUE THREAD

**Issue by erictakahashi** [nvaccess/nvda#8394]
### Question:
- For some `div`s, the **NVDA** tool is reading the entire content of the `div`, which is not user friendly depending on the content. I tried to to set **aria-label** for the `div`, but still didn't work. 
- Is there a way to determine a specific text to be read for a certain `<div>`?

**Comment by dkager** [comment]
> For some divs, the NVDA tool is reading the entire content of the div

What do you mean? NVDA should only read up to the configured line length in browse mode, or read the element by line or paragraph. In the end, the user should have access to the full text contained in the element.

**Comment by ehollig** [comment]
Hello @erictakahashi,

What "role" do you have set for the div?
Does this div have a tabindex?
Do you have example code of the div that you could provide, along with the steps that you took to reproduce the issue you are experiencing?

**Comment by erictakahashi** [comment]
Hi @ehollig, the role is `dialog`.
There's an example on this page https://jsfiddle.net/6gq9x/37/
If you click on the "launch modal" button, a dialog will open.
Notice on the code that there's an `aria-labelledby` set, but the screen reader is speaking not only the modal title but also the buttons, and for this case I don't want it to speak the buttons unless I focus on them.

**Comment by ehollig** [comment]
@erictakahashi, you may have "Automatic Say All on page load" option in the Browse Mode settings enabled. When this option is enabled, NVDA reads the entire page, starting from the beginning, when a new page or modal opens. Unchecking this option only reads the modal title. This option is enabled by default, but you can disable it by pressing NVDA+CTRL+B and unchecking the "Automatic Say All on page load" option.

**Comment by erictakahashi** [comment]
HI @ehollig, this does not avoid the reader to announce some other elements inside the dialog, but it improved a lot. Thank you very much.
